src/sentry/api/endpoints/organization_release_previous_commits.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+
+from sentry.api.bases.organization import OrganizationReleasesBaseEndpoint
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.serializers import serialize
+from sentry.models import Release
+
+
+class OrganizationReleasePreviousCommitsEndpoint(OrganizationReleasesBaseEndpoint):
+    def get(self, request, organization, version):
+        """
+        Retrieve an Organization's Most Recent Release with Commits
+        ````````````````````````````````````````````````````````````
+
+        :pparam string organization_slug: the slug of the organization the
+                                          release belongs to.
+        :pparam string version: the version identifier of the release.
+        :auth: required
+        """
+
+        try:
+            release = Release.objects.get(organization_id=organization.id, version=version)
+        except Release.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        if not self.has_release_permission(request, organization, release):
+            raise ResourceDoesNotExist
+
+        start_date = release.date_released or release.date_added
+
+        prev_release_with_commits = (
+            Release.objects.filter(
```

Comment:
This will check literally all releases for all projects related to that release, every time. I do worry that this may be inefficient for cases where the most recent previous release has the commit. However, I don't know how often this will be used (by how many orgs and also how often will they be creating new releases).